### DISCUSSION TITLE
How to deploy the nextjs project on windows server?

### DISCUSSION THREAD

**Discussion by jxnx888** [vercel/next.js#36868] [Help] [answered]
As for the topic, How do I deploy the nextjs project? I am new to the deployment.

I have already installed Node.js on my server. 

1. what kind of files do I need to copy to the server? 
    a. The whole project?
    b. the .next folder?
    c.  or out folder?
3. Please share the specific steps with me, I really appreciate it.

**Comment by icyJoseph** [comment, marked as answer]
Hi,

If we follow the docs, the output of a build is in `.next`. However if you are using `next export` after running `next build`, then you must use `out`.

- `.next` to run a server
- `out` if you are doing a static HTML site

I am assuming you are running a server.

Copying `.next` might not be so easy, because it is a lot of files. Often anti-virus want to scan every single file during copying. You might also look track of which commit are you deploying.

- I'd recommend, installing `git` on that server, and give it whatever credentials it needs to only clone the project (read only)
- Install dependencies
- If you have written unit/integration/e2e tests, it is a good idea to run them now
- Build the project on the server
- Start the server with `next start`
- You should make sure that the port where the server starts is reachable

I think this page has some good information about Next.js specifics https://nextjs.org/docs/deployment

**Reply by jxnx888** [reply]
Thanks, @icyJoseph, I was trying to only use `.next` and copy the package.json to `npm I && npm run start`, but it caused all page 404 even though I do not have the capitalized name/path. 

I would like to try to clone the whole project to the server, however, I don't think it is a great way to have the whole project on the server.  In my previous experience, we only need to keep the built folder on the server.
